### PR TITLE
Adds optional extra args to the  util

### PR DESCRIPTION
**Pull request by glyg**
Follow up on #235 

The extra test is very minimal for now, I wanted to get your input before doing more.

I choose to put it in `test_import.py` to avoid rewriting all the mock & setup machinery, but I'm not sure it's ok / good practice...

### CODE REVIEW

File: src/omero/util/import_candidates.py [hunk 1]
```diff
@@ -29,43 +29,64 @@ def _to_list(path):
...
+    """
+    if extra_args is None:
+        extra_args = []
     path = _to_list(path)
     readers = str(readers)
     cli = CLI()
     cli.loadplugins()
     if readers:
-        cli.invoke(["import", "-l", readers, "-f"] + path)
+        cli.invoke(["import", "-l"] + extra_args + [readers, "-f"] + path)
```
Comment (joshmoore) [inline src/omero/util/import_candidates.py:50]:
```suggestion
        cli.invoke(["import", "-l", readers] + extra_args + ["-f"] + path)
```

File: src/omero/util/import_candidates.py [hunk 2]
```diff
@@ -29,43 +29,64 @@ def _to_list(path):
...
+    ...
+    last_file_in_group
+    ],
+    some_file_in_second_group : ...
+    }
+
+    you can pass more arguments to the `import` command through the
+    extra_args argument in the form of a list.
+
+    to_import = as_dictionnary("/my/dir/with_tifs", extra_args=["--depth", "6"])
```
Comment (joshmoore) [inline src/omero/util/import_candidates.py:76]:
```suggestion
    to_import = as_dictionary("/my/dir/with_tifs", extra_args=["--depth", "6"])
```

File: src/omero/util/import_candidates.py [hunk 3]
```diff
@@ -29,43 +29,64 @@ def _to_list(path):
...
+
+    will go through the directories with a depth level of 6 instead of
+    the default 4.  Arguments of interest might be: `debugging`,
+    `report`, `logback`. Note that the command is ran localy so options
```
Comment (joshmoore) [inline src/omero/util/import_candidates.py:80]:
```suggestion
    `report`, `logback`. Note that the command runs locally so options
```

File: test/unit/clitest/test_import.py [hunk 4]
```diff
@@ -512,3 +514,12 @@ def do_import(self, command_args, xargs, mode):
         self.args = ["mock-import", "-f", "---bulk=%s" % b]
         self.add_client_dir()
         self.cli.invoke(self.args, strict=True)
+
+    @pytest.mark.skipif(sys.platform == "win32", reason="Fails on Windows")
```
Comment (joshmoore) [inline test/unit/clitest/test_import.py:518]: What's the failure on Windows? Something we can help with?

Comment (glyg) [reply test/unit/clitest/test_import.py:518]: The Travis traceback is [here](https://travis-ci.org/github/ome/omero-py/jobs/716528635#L3149)

```
E           omero.InternalException: exception ::omero::InternalException
E           {
E               serverStackTrace = None
E               serverExceptionClass = None
E               message = 'import -f' exited with a rc=108. See console for more information
E           }
```

Most of the tests in the class are decorated by the same skip instruction, maybe a filesystem issue?

**Comment by glyg** [comment]
I'll add test to check that the extra args are included correctly

**Comment by glyg** [comment]
sorry for the messy diff, I passed black on the file automatically, I can revert that if you want. I don't know if further tests / doc are needed?

**Comment by joshmoore** [comment]
Yeah, let's force push away the application of black here. We'll do that in concert with a GitHub Action to make sure it stays formatted.

Otherwise, test looks good. Thanks.

cc: @sbesson

**Comment by sbesson** [comment]
Thanks @glyg, merging for inclusion in the upcoming minor release of `omero-py`

**Comment by glyg** [comment]
Great! Thanks to both of you!

**Pull request merged**